src/serialization/sb3.js

```diff
@@ -690,6 +690,7 @@ const deserializeInputs = function (inputs, parentId, blocks) {
     for (const inputName in inputs) {
         if (!inputs.hasOwnProperty(inputName)) continue;
         const inputDescArr = inputs[inputName];
+        if (!Array.isArray(inputDescArr)) continue;
```

Comment:
I think I'm still a little hesitant about this change. 

At the very least, this change (and the similar one below) needs a comment to explain that this line is solely for accommodating deserializing deserialized blocks (which is not the normal use case of this function).

In general, I think it's good for functions like this to be idempotent, but one thing that's weird about making deserialize work on already deserialized things is that there isn't really a similar analogy for `serialize` to work that way.

The closest (albeit somewhat different) thing to this is JSON.parse and JSON.stringify... 
`JSON.stringify(JSON.stringify(x))` and `JSON.stringify(x)` are not the same thing, and `JSON.parse(JSON.parse(x))` doesn't work.

Reply:
basically - there is no way to install targets from json without getting passed through this `deserialize`, I think it makes sense to make `deserialize(alreadySerialized)` result in no change (as opposed to weird undefined variables and broken code as it does with the current setup)

Reply:
Do you mind adding comments above both of those lines since they are not required for the normal/common usage flow of deserialize?